presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java

```diff
@@ -366,17 +366,18 @@ private ValuesNode values(int planNodeId, String... symbols)
     private class FixedLookup
             implements Lookup
     {
-        private Map<PlanNodeId, PlanNodeCostEstimate> costs;
-        private Map<PlanNodeId, PlanNodeStatsEstimate> stats;
+        private Function<String, PlanNodeCostEstimate> costs;
+        private Function<String, PlanNodeStatsEstimate> stats;
+
+        public FixedLookup(Function<String, PlanNodeCostEstimate> costs, Function<String, PlanNodeStatsEstimate> stats)
+        {
+            this.costs = costs;
+            this.stats = stats;
+        }
 
         public FixedLookup(Map<String, PlanNodeCostEstimate> costs, Map<String, PlanNodeStatsEstimate> stats)
         {
-            this.costs = costs.entrySet()
-                    .stream()
-                    .collect(toMap(entry -> new PlanNodeId(entry.getKey()), Map.Entry::getValue));
-            this.stats = stats.entrySet()
-                    .stream()
-                    .collect(toMap(entry -> new PlanNodeId(entry.getKey()), Map.Entry::getValue));
+            this(costs::get, stats::get);
```

Comment:
> Are you sure that previous implementation wasn't copying the data purposely?

I think it's just been mapping to `PlanNodeId`-s. All the maps passed are immutable, and if they weren't and s/b would mutate the maps during the tests, there's a special place in hell for them.

>instead adding a layer of functional-hocus-pocus

The hocus-pocus is b/c for testing the 'no stats' scenario, I need the maps to return an `UNKNOWN_STATS/COSTS` for every possible node. I'd prefer not passing all the node ids, especially that once we need to add more nodes, that's another place we need to pass them to.